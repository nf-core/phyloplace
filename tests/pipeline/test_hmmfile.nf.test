nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_hmmfile"
    tag "pipeline"

    test("HMM file align command line params") {
        config "./nextflow.config"
        when {
            params {
                outdir          = "$outputDir"
                queryseqfile    = "https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/PF14720_3_sequences.faa"
                refseqfile      = "https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/PF14720_seed.faa"
                hmmfile         = "https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/PF14720.hmm"
                refphylogeny    = "https://github.com/nf-core/test-datasets/raw/phyloplace/testdata/PF14720_seed.ft.LGCAT.newick"
                model           = "LG"
                taxonomy        = "https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/gappa/gappa_taxonomy.tsv"
                max_cpus        = 2
                max_memory      = '6.GB'
                max_time        = '6.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/pipeline_info/nf_core_phyloplace_software_mqc_versions.yml").exists() },
                { assert new File("$outputDir/epang/placement.epa_result.jplace.gz").exists() },
                { assert snapshot(path("$outputDir/gappa/placement.graft.placement.epa_result.newick")).match("placement.graft.placement.epa_result.newick") },
                { assert snapshot(path("$outputDir/gappa/placement.heattree.colours.txt")).match("placement.heattree.colours.txt") },
                { assert snapshot(path("$outputDir/gappa/placement.heattree.tree.nexus")).match("placement.heattree.tree.nexus") },
                { assert snapshot(path("$outputDir/gappa/placement.heattree.tree.phyloxml")).match("placement.heattree.tree.phyloxml") },
                { assert new File("$outputDir/gappa/placement.heattree.tree.svg").exists() },
                { assert new File("$outputDir/gappa/placement.taxonomy.krona.profile").exists() },
                { assert snapshot(path("$outputDir/gappa/placement.taxonomy.labelled_tree.newick")).match("placement.taxonomy.labelled_tree.newick") },
                { assert new File("$outputDir/gappa/placement.taxonomy.per_query.tsv").exists() },
                { assert new File("$outputDir/gappa/placement.taxonomy.profile.tsv").exists() },
                { assert new File("$outputDir/gappa/placement.taxonomy.sativa.tsv").exists() },
                { assert snapshot(path("$outputDir/hmmer/placement.query.hmmalign.masked.afa.gz")).match("placement.query.hmmalign.masked.afa.gz") },
                { assert snapshot(path("$outputDir/hmmer/placement.ref.hmmalign.masked.afa.gz")).match("placement.ref.hmmalign.masked.afa.gz") },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
            )
        }
    }
}
