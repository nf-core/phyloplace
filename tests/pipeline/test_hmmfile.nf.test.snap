{
    "placement.heattree.tree.phyloxml": {
        "content": [
            "placement.heattree.tree.phyloxml:md5,ea35b07e4fd8d174fd6b2f204629cc4e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T14:52:12.135664245"
    },
    "placement.heattree.colours.txt": {
        "content": [
            "placement.heattree.colours.txt:md5,7704cd341c8693981253ba6d740c1061"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T14:52:12.118678241"
    },
    "placement.heattree.tree.nexus": {
        "content": [
            "placement.heattree.tree.nexus:md5,e1cfdcb63f3a751677d0e78fb6b9230a"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T14:52:12.126685402"
    },
    "placement.ref.hmmalign.masked.afa.gz": {
        "content": [
            "placement.ref.hmmalign.masked.afa.gz:md5,3452f0d4646209116c3b60f1c77d5ba3"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T14:52:12.177027699"
    },
    "placement.query.hmmalign.masked.afa.gz": {
        "content": [
            "placement.query.hmmalign.masked.afa.gz:md5,69b6d82789587f368da4d808ce20744a"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T14:52:12.165540396"
    },
    "placement.graft.placement.epa_result.newick": {
        "content": [
            "placement.graft.placement.epa_result.newick:md5,1b9415cffdc5b744e7339be64436932e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T14:52:12.104259806"
    },
    "placement.taxonomy.labelled_tree.newick": {
        "content": [
            "placement.taxonomy.labelled_tree.newick:md5,39de8a46303ea757f6eb94478db1ffe0"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-16T14:52:12.150397494"
    }
}